runtime/drivers/file/model_executor_olap_self.go

```diff
@@ -328,22 +339,54 @@ func writeParquet(res *drivers.Result, path string) error {
 				recordBuilder.Field(i).(*array.BinaryBuilder).Append(v)
 			}
 		}
+		rows++
+		// We write the data to the underlying file to not buffer entire data in memory till it is written to parquet file.
+		// Since the check is not on memory but on number of rows this can still consume lot of memory
+		// so we set the limit to a much aggressive value than the default which is 64M.
+		if rows == maxParquetRowGroupLen {
```

Comment:
Re comment above – I think this is too aggressive. Have you looked into whether there are any settings/stats on the options on `NewFileWriter` and/or on `parquetwriter` that can be used to solve this? Maybe `RowGroupTotalCompressedBytes`? 

It's okay to go a row group size over in terms of memory usage. As long as we enforce the right limit on the export file, and we don't use unbounded memory.

Reply:
The problem is not with the `parquetwriter`. The problem is that the entire data is kept in-memory in form of `arrow.Record` till it is written to the file using the writer. 
If you feel that our limits are too aggressive (though it is still higher than the defaults use by clickhouse and duckdb) we can use the following approach :
1. Write data to the file in batches of 1000 using the `WriteBuffered` API which keeps the data in-memory (I **think** there should not be a copy because it should use the underlying `arrow.Records` which is the main USP of arrow :) )
2. Keep checking the `RowGroupTotalCompressedBytes` to see what is the size of the data written so far and if it exceeds the limit throw an error.
3. If the limit is not exceed but the total bytes exceed limit of `512MB` create a new row group.

Reply:
I'm not too worried about the aggressive limit, though it would be nice to stick with the defaults, I'm more worried about the potential issues relating to `Since the check is not on memory but on number of rows this can still consume lot of memory`. We're moving away from row-based limits to byte-based limits due to those issues. Some people have very wide rows (especially in a pivoted response) that might have large binary fields.

Reply:
Cool. In that case I think the solution proposed above can fix that issue.